src/generators.rs

```diff
@@ -2213,6 +2213,95 @@ pub fn lollipop_graph(
     Ok(graph)
 }
 
+/// Generate a generalized Petersen graph :math:`G(n, k)` with :math:`2n`
+/// nodes and :math:`3n` edges. See Watkins [1]_ for more details.
+///
+/// .. note::
+///   
+///   The Petersen graph itself is denoted :math:`G(5, 2)`
+///
+/// :param int n: number of nodes in the internal star and external regular polygon.
+/// :param int k: shift that changes the internal star graph.
+/// :param bool multigraph: When set to False the output
+///     :class:`~retworkx.PyGraph` object will not be not be a multigraph and
+///     won't allow parallel edges to be added. Instead
+///     calls which would create a parallel edge will update the existing edge.
+///
+/// :returns: The generated generalized Petersen graph.
+///
+/// :rtype: PyGraph
+/// :raises TypeError: If either ``n`` or ``k`` are
```

Comment:
Maybe?
```suggestion
/// :raises IndexError: If either ``n`` or ``k`` are
```
Or even better we can mention that `TypeError` will be raised if you pass an arg that can not be converted to a `usize` and `IndexError` if the value is not "valid".

Reply:
Done